### ISSUE TITLE
scatter_logsumexp producing warnings on tensors which require gradients

### ISSUE THREAD

**Issue by SergioRAgostinho** [rusty1s/pytorch_scatter#167]
Just reporting a cumbersome warning that has been creeping up my logs :)

torch_scatter version: 2.0.5
torch version: 1.5.1

```python
import torch
from torch_scatter import scatter_logsumexp

data = torch.rand(10)
data.requires_grad = True
batch = torch.zeros_like(data).long()

# the culprit. produces
# /opt/conda/conda-bld/pytorch_1591914855613/work/torch/csrc/autograd/custom_function.cpp:61: UserWarning: Only input Tensors should be given to ctx.mark_dirty(). If a Tensor is not an input, there is no need to pass it to mark_dirty().
scatter_logsumexp(data, batch)
